chain/default_syncer.go

```diff
@@ -179,16 +179,16 @@ func (syncer *DefaultSyncer) tipSetState(ctx context.Context, tsKey string) (sta
 // Precondition: the caller of syncOne must hold the syncer's lock (syncer.mu) to
 // ensure head is not modified by another goroutine during run.
 func (syncer *DefaultSyncer) syncOne(ctx context.Context, parent, next types.TipSet) error {
-	head := syncer.chainStore.Head()
+	head := syncer.chainStore.GetHead()
```

Comment:
i'm a little iffy on how this works, but it must because locking behavior hasn't changed but the caller has an exclusive lock, how are we getting a reader lock in gethead?

Reply:
a lock was added on lines 366 & 367